### ISSUE TITLE
window is not defined using vuex-persist and NUXT

### ISSUE THREAD

**Issue by warheartvik** [championswimmer/vuex-persist#88]
Hello,

I followed the instructions on https://www.npmjs.com/package/vuex-persist. But I got 'window is not defined' error. I'm using nuxt and typescript

**Comment by kpturner** [comment]
Bit difficult to follow ^^ because you haven't formatted the code as 
`this is code` above, but anyway I was looking to use this plug-in with Nuxt (typescript) also but I am a bit reticent until you get an answer. Obviously `window.anything` isn't going to fly when executed server side.  It does beg the question for me as somewhat of a n00b as to where the store exists on the server-side anyway.  Clearly when server-side rendering is taking place the vuex store can be accessed and populated but I have no clue how it is synced between the client and the server.

**Comment by championswimmer** [comment]
Use it as a client only plugin, as shown in README

Yes the store values can only be used in a hydrated app (one that has
become reactive after loading on the frontend)

**Comment by kpturner** [comment]
Isn't the OP using it is a client-only plugin then?  The README says to use `ssr: false` and he is - but still gets the `window is not defined` error.

**Comment by kpturner** [comment]
OK I see what he did wrong. I have it at least storing state in localStorage now. I have another issue that I will log separately.

**Comment by kyriediculous** [comment]
Note that no-ssr still "renders" the content internally (but doesn't render the html, not until the parent is mounted).

**Comment by nasermirzaei89** [comment]
I used this code:
```ts
function getPlugins() {
    let plugins = []

    if (process.browser) {
        const vuexLocal = new VuexPersistence<RootState>({
            storage: window.localStorage
        })

        plugins.push(vuexLocal.plugin)
    }
}

export const plugins = getPlugins()
```

So, I get rid of error `window is not defined`

**Comment by topzdev** [comment]
@nasermirzaei89  you forgot to return the plugins

**Comment by nasermirzaei89** [comment]
@topzdev  as I see in **nuxt.js** guide, we should only export it as `plugins`:

https://nuxtjs.org/guide/vuex-store/#plugins
